content/articles/searching-1.md

```diff
@@ -0,0 +1,257 @@
+__Mom:__ (_held it up to my face_) That’s because you’ve been looking for it
+with your nose.
+
+We all know this story, we’ve experienced this at least a few times in our
+lives, and it is to show that many Moms (or Dads) are the best search and
+analytics engine. So why am I telling you this story?
+
+Search functionality can be complex, and search UI can be complicated. To help
+design and develop the search feature, we need user stories and diagrams, and
+to lead us to these deliverables, we should ask a few guiding questions. Let’s
+consider that universal story again to help us deconstruct and reassemble our
+approach to search functionality.
+
+### Who am I: Who is the user? ###
+The persona of the user will guide us in deciding what the
+[principle of least astonishment is](https://en.wikipedia.org/wiki/Principle_of_least_astonishment)
+for the sets of users, and inform us how we should design the UI for the search
+functionality. What can we assume about their experience with web searching?
+What is the basis of our assumptions? What devices do they use? Will, or should
+search be available on all devices? Knowing the user will lead us to __what__,
+__where__, and __how__ they would want to search for things.
+
+### What is the user searching for? ###
+To put simply, the user is searching for an object. But what attributes define
+the object? Are there relationships with other objects? Are these relationships
+relevant to the search? In the “Green Notebook Quest” story, the object that I
+was looking for can initially be broken down as follows:  
+`type: notebook`  
+`color: green`
+
+And as it turned out, there were more than one green notebook.
+
+<div class="row">
+<div class="col-sm-3">
+<img src="/img/assets/notebook-lg.png">
+</div>
+<div class="col-sm-3">
+<img src="/img/assets/notebook-md.png">
+</div>
+</div>
+
+But then, in the midst of frustration, I expanded my query to include other
+attributes, such as the object’s dimensions (“smaller”), what it’s being used
+for, or title, (“Math”), and tags (“cat sticker”): 
+
+<div class="row">
+<div class="col-sm-4 text-center">
+<img src="/img/assets/notebook-lg.png">
+<p><i class="fa fa-times-circle fa-3x text-muted" aria-hidden="false"></i></p>
+</div>
+<div class="col-sm-8">
+<code>
+type: notebook<br />
+color: green<br />
+title: "English Comp"<br />
+tag: <br />
+size: "large"<br />
+width: 9in<br />
+height: 14in
+</code>
+</div>
+</div>
+<hr />
+<div class="row">
+<div class="col-sm-4 text-center">
+<img src="/img/assets/notebook-md.png">
+<p><i class="fa fa-check-circle fa-3x text-success"
+aria-hidden="false"></i></p>
+</div>
+<div class="col-sm-8">
+<code>
+type: notebook<br />
+color: green<br />
+title: "Math"<br />
+tag: cat<br />
+size: "small"<br />
+width: 7in<br />
+height: 11in
+</code>
+</div>
+</div>
+
+From this example, we see that it is helpful to understand the definition of
+the object the is searching for, in any given application, so that we can model
```

Comment:
we see that it is helpful to understand the definition of the object `we are searching for` or `the user is searching for`.